pkgs/by-name/eu/eudic/package.nix

```
@@ -0,0 +1,104 @@
+{ fetchurl
+, stdenv
+, autoPatchelfHook
+, makeWrapper
+, lib
+, makeDesktopItem
+, copyDesktopItems
+, fontconfig
+# Dependences of applications declaim
+, libnotify
+, libX11
+, libXScrnSaver
+, libXext
+, libXtst
+, libuuid
+, libsecret
+, xdg-utils
+, xdg-utils-cxx
+, at-spi2-atk
+# Dependences of autoPatchelfHook
+, gtk3
+, alsa-lib
+, libdrm
+, e2fsprogs
+, nss
+, libgpg-error
+, libjack2
+, mesa
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "eudic";
+  version = "13.5.2";
+
+  src = fetchurl {
+    url = "https://www.eudic.net/download/eudic.deb?v=${finalAttrs.version}";
+    hash = "sha256-UPkDRaqWF/oydH6AMo3t3PUT5VU961EPLcFb5XwOXVs=";
+  };
+
+  nativeBuildInputs = [
+    autoPatchelfHook
+    makeWrapper
+    copyDesktopItems
+  ];
+
+  # Depends: libgtk-3-0, libnotify4, libnss3, libxss1, libxtst6, xdg-utils, libatspi2.0-0, libuuid1, libsecret-1-0
+  buildInputs = [
+    libnotify
+    libX11
+    libXScrnSaver
+    libXext
+    libXtst
+    libuuid
+    libsecret
+    xdg-utils
+    xdg-utils-cxx
+    at-spi2-atk
+    # Dependences base on autoPatchelfHook and nix-locate
+    gtk3                # libEGL.so.1 libfreetype.so.6 libfontconfig.so.1
+    alsa-lib            # libasound.so.2
+    e2fsprogs           # libcom_err
+    nss                 # libnss3.so libnssutils3.so
+    libgpg-error        # libgpg-error.so
+    libjack2            # libjack2.so.0
+    mesa                # libgbm.so.1
+  ];
+
+  unpackPhase = ''
+    ar x $src
+    tar xf data.tar.xz
+  '';
+
+  # After unpackPhase
+  # File is /usr/share/{Application, eusoft-eudic, icons, pixmaps}
+  installPhase = ''
+    runHook preInstall
+
+    # Copy & Paste Original File
+    echo $out
+    mkdir -p $out
+    cp -r usr/* $out/
+
+    # Build Wrapper
+    makeWrapper $out/share/eusoft-eudic/eudic $out/bin/eudic \
+      --argv0 "eudic" \
```

```suggestion
      --inherit-argv0 \
```
https://github.com/NixOS/nixpkgs/blob/febf0c45943643f076d39bce6cd6f90ce20b2eaa/pkgs/build-support/setup-hooks/make-wrapper.sh#L16C3-L16C18

---

➜  nixpkgs git:(new-eudic) ✗ nix-build . -A eudic 
this derivation will be built:
  /nix/store/swl03lq4jhdg32p9z5vpdf7bvppp253h-eudic-13.5.2.drv
building '/nix/store/swl03lq4jhdg32p9z5vpdf7bvppp253h-eudic-13.5.2.drv'...
Running phase: unpackPhase
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
no Makefile or custom buildPhase, doing nothing
Running phase: glibPreInstallPhase
Running phase: installPhase
/nix/store/k34h1c9qxrp9x006kdbimdp1fvww4cjy-eudic-13.5.2
/nix/store/d3dzfy4amjl826fb8j00qp1d9887h7hm-stdenv-linux/setup: line 1590: --inherit-argv0: command not found
error: builder for '/nix/store/swl03lq4jhdg32p9z5vpdf7bvppp253h-eudic-13.5.2.drv' failed with exit code 127;
       last 10 log lines:
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > no configure script, doing nothing
       > Running phase: buildPhase
       > no Makefile or custom buildPhase, doing nothing
       > Running phase: glibPreInstallPhase
       > Running phase: installPhase
       > /nix/store/k34h1c9qxrp9x006kdbimdp1fvww4cjy-eudic-13.5.2
       > /nix/store/d3dzfy4amjl826fb8j00qp1d9887h7hm-stdenv-linux/setup: line 1590: --inherit-argv0: command not found
       For full logs, run 'nix log /nix/store/swl03lq4jhdg32p9z5vpdf7bvppp253h-eudic-13.5.2.drv'.